packages/core/src/StateNode.ts

```diff
@@ -44,14 +45,17 @@ import { StateMachine } from './StateMachine';
 
 const EMPTY_OBJECT = {};
 
-interface StateNodeOptions<TContext, TEvent extends EventObject> {
+interface StateNodeOptions<
+  TContext extends MachineContext,
+  TEvent extends EventObject
+> {
   _key: string;
   _parent?: StateNode<TContext, TEvent>;
   _machine: StateMachine<TContext, TEvent>;
 }
 
 export class StateNode<
-  TContext = any,
+  TContext extends MachineContext = {},
```

Comment:
Shouldn't it be like this?
```suggestion
  TContext extends MachineContext = MachineContext,
```